### PR TITLE
NR-31202 C# integration

### PR DESCRIPTION
**Pull request by pkosanam**
# Summary

Ticket URL: https://issues.newrelic.com/browse/NR-31202

Description: 
This PR has the new quickstart integration named "C#"

A concise description of the changes being introduced. If you are submitting a PR for a
new quickstart please review the pre merge checklist section to validate this PR is ready for review.

## Pre merge checklist

- [ ] Did you check you NRQL syntax? - Does it work?
- [ ] Did you check your dashboard image quality? -  Do they look good?
- [ ] Did you check that your alerts actually work?
- [ ] Did you include an InstallPlan and Documentation reference?
- [ ] Did you check your descriptive content for voice, tone, spelling and grammar errors?
- [ ] Did you attach images of your dashboards to the PR so we can see them working?

### Screenshots

Attach images of any visual changes, such as a dashboard here.

### Preview
c-sharp-preview

### CODE REVIEW

**Comment by pkosanam** [comment]
Hi @aswanson-nr ,

All checks on this PR are successful. But there is an issue while viewing the generated preview link.

On trying to view the preview link, the page redirects back to home page https://newrelic.com/instant-observability/

Link generated: https://newrelic.com/instant-observability/preview?local=true&quickstart=dotnet/c#

Screenshot:
image

Could you please help me with this ?

Thank you. 😊

**Comment by aswanson-nr** [comment]
Hey @pkosanam, I've had the terminal cut off the portion of the url that is being broken onto the second line there. Could you confirm that you're getting the whole URL for the preview? It will redirect to the main catalogue if anything is miss configured.

File: quickstarts/dotnet/c#/config.yml [hunk 1]
```diff
@@ -0,0 +1,68 @@
+id: 
+slug: c#
```
Comment (aswanson-nr) [inline quickstarts/dotnet/c#/config.yml:2]:
```suggestion
slug: c-sharp
```

File: quickstarts/dotnet/c#/config.yml [hunk 2]
```diff
@@ -0,0 +1,68 @@
+id:
```
Comment (aswanson-nr) [inline quickstarts/dotnet/c#/config.yml:1]:
```suggestion
```

File: quickstarts/dotnet/c#/config.yml [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+keywords:
+  - apm
+  - dotnet
+  - c#
+  - c sharp
+installPlans:
+  - dotnet-agent
+
+dashboards:
+  - c#
```
Comment (aswanson-nr) [inline quickstarts/dotnet/c#/config.yml:66]:
```suggestion
  - c-sharp
```

File: dashboards/c#/c#.json [hunk 4]
```diff
@@ -0,0 +1,888 @@
+{
```
Comment (aswanson-nr) [inline dashboards/c#/c#.json:1]: We'll need to update the name of this file as well! Something like "dashboards/c-sharp/dashboard.json" should work fine.

**Review by aswanson-nr — COMMENTED** [review]
Hey @pkosanam,
I dug into this a bit more and the issue appears to be coming from the `#` character in the name and file paths. Overall, we do not support special characters in the `slug` field and the file paths. If you take a look at the .NET quickstart, you'll see that everything is labeled `dotnet` except for the `title` field that can be set to any string.
I tossed some comments on lines that should most likely get updated with this change!

File: dashboards/c#/c#.json (hunk 4 above)
Comment (pkosanam) [reply dashboards/c#/c#.json:1]: Thank you Alec, It worked.. 😀

File: quickstarts/dotnet/c#/config.yml (hunk 3 above)
Comment (pkosanam) [reply quickstarts/dotnet/c#/config.yml:66]: Thank you Alec, It worked.. 😀

File: quickstarts/dotnet/c#/config.yml (hunk 2 above)
Comment (pkosanam) [reply quickstarts/dotnet/c#/config.yml:1]: I have seen that, even if the "id" is left blank, the preview worked. 👍

Comment (pkosanam) [reply quickstarts/dotnet/c#/config.yml:1]: Also, I have removed the "id" field and then pushed my changes 👍

File: quickstarts/dotnet/c#/config.yml (hunk 1 above)
Comment (pkosanam) [reply quickstarts/dotnet/c#/config.yml:2]: worked 😃

**Comment by pkosanam** [comment]
c-sharp-preview

File: dashboards/c-sharp/dashboard.json [hunk 5]
```diff
@@ -0,0 +1,888 @@
+{
```
Comment (aswanson-nr) [inline dashboards/c-sharp/dashboard.json:1]: I'm seeing a lot of entity GUIDs and Ids in this file, let's get those removed in favor of more general queries!

**Review by aswanson-nr — CHANGES_REQUESTED** [review]
The quickstart looks great! However, most of the dashboard NRQL queries are making use of information that is unique to an account's entity. We'll need to update these queries to support anyone's entities.

File: quickstarts/dotnet/c-sharp/config.yml [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+  - Learn the whole story of performance errors, from beginning to end
+  - Use traces to connect latency issues to errors, getting you to the root cause as quickly as possible
+  - Use traces to view filename and line number of the C# stack. To reduce noise, logically filter and aggregate C# exceptions
+  - Scalable error monitoring without affecting production flow
+  - Maintain low latency with fast throughput
+  - Get a bird's eye view of your C# applications using New Relic’s easy to use dashboard
+
+  ## Get end-to-end visibility into your C# operation terrain and all the dependencies
+  
+  Automatically discover operation topology and interdependencies, relate and trace crucial business deals grounded on product operation address feature and prioritize the end-to-end business deals performance, not just the health of the usage and structure lumps
```
Comment (akristen) [inline quickstarts/dotnet/c-sharp/config.yml:22]:
```suggestion
Get an overview of your entire stack and trace critical business transactions so you're always delivering the best user experience```

**Comment by pkosanam** [comment]
Closing this PR as we have raised a new PR: https://github.com/newrelic/newrelic-quickstarts/pull/1357

**Pull request closed without merging**